### PR TITLE
Fused LSTM grad-grad

### PR DESCRIPTION
**Pull request by unnonouno**
LSTM grad-grad calls too many kernels. I used `cupy.fuse` to combine them.
Merge #3206 first.

### CODE REVIEW

**Comment by okuta** [comment]
jenkins, test this please.

File: chainer/functions/activation/lstm.py [hunk 1]
```diff
@@ -195,51 +200,63 @@ def backward(self, inputs, grads):
...
+                c_prev, a, i, f, o, c, gc, gh, ggc_prev, gga, ggi, ggf, ggo)
         return gc_prev, gx, gc_next, ggc, ggh
 
 
+def _cupy_sigmoid(x):
+    half = x.dtype.type(0.5)
+    return cuda.fusion.tanh(x * half) * half + half
+
+
+@cuda.fuse()
```
Comment (okuta) [inline chainer/functions/activation/lstm.py]: How about using `input_num`?
`lstm_grad_grad` will accept writable arguments.

```python
@cuda.fuse(input_num=13)
def lstm_grad_grad(
        c_prev, a, i, f, o, c, gc, gh, ggc_prev, gga, ggi, ggf, ggo,
        gc_prev, ga, gi, gf, go, gc_next, ggc, ggh):
```

Comment (unnonouno) [reply chainer/functions/activation/lstm.py]: How to fix?

Comment (okuta) [reply chainer/functions/activation/lstm.py]: I wrote sample. That kernel will reduce array copy.

Comment (okuta) [reply chainer/functions/activation/lstm.py]: Sorry, This is my mistake.

**Comment by okuta** [comment]
LGTM!

**Pull request merged**